src/main/java/seedu/address/model/course/UniqueCourseList.java

```diff
@@ -1,23 +1,25 @@
 package seedu.address.model.course;
 
+import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 import static seedu.address.model.course.Course.MESSAGE_CONSTRAINTS;
 import static seedu.address.model.course.Course.isValidCourseCode;
 
 import java.util.Iterator;
+import java.util.List;
 
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 import seedu.address.model.course.exceptions.CourseNotFoundException;
 import seedu.address.model.course.exceptions.LessonNotFoundException;
+import seedu.address.model.util.SampleDataUtil;
 
 /**
  * Represents a list of courses with TAs in SOC. Immutable.
  */
 public class UniqueCourseList implements Iterable<Course> {
     private static final ObservableList<Course> internalList = FXCollections.observableArrayList(
```

Comment:
This list should not be static under the new implementation. The main inconvenience would be that findByCourseCode would need to be drilled in order to be used.